scripts/install_third_party.py

```diff
@@ -128,9 +129,9 @@ def download_and_untar_files(
         print 'Downloading and untarring file %s to %s' % (
             tar_root_name, target_parent_dir)
         common.ensure_directory_exists(target_parent_dir)
-
         urllib.urlretrieve(source_url, TMP_UNZIP_PATH)
-        with tarfile.open(TMP_UNZIP_PATH, 'r:gz') as t:
+	print TMP_UNZIP_PATH
+        with closing(tarfile.open(TMP_UNZIP_PATH, 'r:gz')) as t:
```

Comment:
I am not sure either but when I tested it once, it ran fine. When I removed
the third party libraries and tested again it threw the error. It hit numpy
first because numpy was the first call to download_and_untar_files but when
I moved a different call to the top of that section it also threw the error
for gae_mapreduce. I plan on testing on a few different Python
installations.

On Thu, Nov 5, 2015 at 6:44 PM, Sean Lip <notifications@github.com> wrote:

> In scripts/install_third_party.py
> <https://github.com/oppia/oppia/pull/1104#discussion_r44086628>:
>
> >          urllib.urlretrieve(source_url, TMP_UNZIP_PATH)
> > -        with tarfile.open(TMP_UNZIP_PATH, 'r:gz') as t:
> > +	print TMP_UNZIP_PATH
> > +        with closing(tarfile.open(TMP_UNZIP_PATH, 'r:gz')) as t:
>
> I like this (well done on finding it!) -- but I'm also confused. Why did
> this trigger just for numpy, and not for the existing third-party libs?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/oppia/oppia/pull/1104/files#r44086628>.
>